From 3599cfa23c3c6d17f72a14278075678e426bf4bf Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Mon, 3 Nov 2008 16:38:50 +0800
Subject: [PATCH] enable jiffy clock api for Maple

Add clock api support for Maple: jiffy clock api is selected,
and openpic clock api is disabled.

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/powerpc/kernel/time.c |    3 +++
 arch/powerpc/sysdev/mpic.c |    2 ++
 2 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/time.c b/arch/powerpc/kernel/time.c
index 0c9b3b8..5ae9613 100644
--- a/arch/powerpc/kernel/time.c
+++ b/arch/powerpc/kernel/time.c
@@ -546,6 +546,9 @@ void __init iSeries_time_init_early(void)
 }
 #endif /* CONFIG_PPC_ISERIES */
 
+#if defined(CONFIG_HWTIMER_HOOKS) && (defined(CONFIG_PPC_MAPLE))
+ #define HWTIMER_USE_JIFFY 1
+#endif
 
 #if defined(HWTIMER_USE_JIFFY)
 
diff --git a/arch/powerpc/sysdev/mpic.c b/arch/powerpc/sysdev/mpic.c
index f7ed652..e9518ff 100644
--- a/arch/powerpc/sysdev/mpic.c
+++ b/arch/powerpc/sysdev/mpic.c
@@ -264,6 +264,8 @@ static inline void _mpic_irq_write(struct mpic *mpic, unsigned int src_no,
 #define mpic_irq_read(s,r)	_mpic_irq_read(mpic,(s),(r))
 #define mpic_irq_write(s,r,v)	_mpic_irq_write(mpic,(s),(r),(v))
 
+/* Discard openpic clock api for Maple */
+#undef CONFIG_HWTIMER_HOOKS
 
 #ifdef CONFIG_HWTIMER_HOOKS
 #include <linux/hwtimer.h>
-- 
1.6.3.3

